package vates.login;


import vates.pages.LoginPage;
import vates.pages.PageGeneral;
import org.testng.Assert;
import org.testng.annotations.Test;


public class TestLogin extends PageGeneral {
  @Test(description = "Test de login  Correctamente", testName = "Test login Correcto", groups = {
          TestGroups.LOGIN})
  public void login() throws Exception {
    config();
    LoginPage login = new LoginPage(driver);
    Assert.assertTrue(login.login(),
            "El usurio  no se pudo loguear correctamente...");
    System.out.println("Usuario Logueado");
    sleep(3000);
    login.logout();
  }

  @Test(description = "Test de logout", testName = "Test de logout", groups = {
          TestGroups.LOGIN})
  public void logout() throws Exception {
    config();
    LoginPage page = new LoginPage(driver);
    page.loginManual("standard_user", "secret_sauce");
    sleep(3000);
    page.logout();
    Assert.assertTrue(true);
  }

  @Test(description = "Test de login con usuario incorrecto", testName = "usuario incorrecto", groups = {
      TestGroups.LOGIN})
  public void userLoked() throws Exception {
      config();
    LoginPage page = new LoginPage(driver);
    Assert.assertTrue(page.userLoked("locked_out_user", "secret_sauce"),
        "No se pudo validar el test de usuer incorrecto");
    System.out.println("Usuario bloqueado");
  }

  @Test(description = "Test de login con password incorrecto", testName = "password incorrecto", groups = {
      TestGroups.LOGIN})
  public void password_incorrecto() throws Exception {
      config();
    LoginPage page = new LoginPage(driver);
    Assert.assertTrue(page.userAndPasswordIncorrecto("performance_glitch_user", "ksksksks"),
        "No se pudo validar el test de password incorrecto");
    System.out.println("Contrase√±a incorrecta");
  }
}


